From 8989e0c7eca187545d9867df22fe358ea880ea4e Mon Sep 17 00:00:00 2001
From: Marsell Kukuljevic <marsell@joyent.com>
Date: Wed, 3 Oct 2018 22:14:27 +1300
Subject: [PATCH] OS-7278 'imgadm create' does not copy parent VM's bootrom
 into imgmanifest Reviewed by: Mike Gerdts <mike.gerdts@joyent.com> Approved
 by: Mike Gerdts <mike.gerdts@joyent.com>

---
 src/img/lib/imgadm.js | 10 ++++++++++
 1 file changed, 10 insertions(+)

diff --git a/src/img/lib/imgadm.js b/src/img/lib/imgadm.js
index aad2eebb..cc9e9fd5 100644
--- a/src/img/lib/imgadm.js
+++ b/src/img/lib/imgadm.js
@@ -3493,6 +3493,16 @@ IMGADM.prototype.createImage = function createImage(options, callback) {
                 log.debug({brand: m.requirements.brand},
                     'set image requirements.brand to source VM brand');
             }
+            if (vmInfo.brand === 'bhyve' && vmInfo.bootrom
+                && !(options.manifest.requirements
+                    && options.manifest.requirements.hasOwnProperty('bootrom')))
+            {
+                if (!m.requirements)
+                    m.requirements = {};
+                m.requirements.bootrom = vmInfo.bootrom;
+                log.debug({bootrom: m.requirements.bootrom},
+                    'set image requirements.bootrom to source VM bootrom');
+            }
             if (incremental) {
                 if (!originInfo) {
                     next(new errors.VmHasNoOriginError(vmUuid));
-- 
2.21.0

